### ISSUE TITLE
Tests stop at first failure

### ISSUE THREAD

**Issue by vsanta** [substack/tape#422]
Hello,

Not sure if it is bad configuration/knowledge of the tool on my end but I noticed the test runner stops when it encounters the first failure.

Is this this by design? Is there a way to run all tests and let the report tell me what passes and what doesn't?

**Comment by ljharb** [comment]
That shouldn't be the case; each `test()` or `t.test()` should run until the first failure. What does your test code look like?

**Comment by vsanta** [comment]
At this point it is pretty short, I'll give you a sample.

```
test('Valid token should return successful response', (t) => {
  t.plan(1);

  fetch(API_URL, AUTH0_TOKEN).then((response) => {
    t.ok(response.success, 'Token is valid');
  });
});

test('Invalid token should return unsuccessful response', (t) => {
  t.plan(1);

  fetch(API_URL, 'invalid_token').catch((error) => {
    t.notOk(error.success, 'Token is invalid');
  });
});
```

And we run as `tape src/**/*.test.js | node_modules/.bin/tap-spec`.

What happens is, if `'Valid token should return successful response'` fails, then `'Invalid token should return unsuccessful response'` never runs. The failure also causes other test files to not run.

**Comment by ljharb** [comment]
Interesting, that doesn't seem to be correct to me. Do you see the same behavior if you run the one file with `node` instead of `tape`? Also, do you see the same behavior with `tape` *without* the pipe to tap-spec?

**Comment by vsanta** [comment]
I definitely see the same behavior without the pipe operator.  

re: run with node... I'm not sure what you mean.

For reference, I tried using it with `tape-promise/tape` and got the same behavior.

The output is:
```
# Valid token should return successful response
(node:34108) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): #<Object>
(node:34108) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
not ok 1 plan != count
  ---
    operator: fail
    expected: 1
    actual:   0
  ...
not ok 2 test exited without ending
  ---
    operator: fail
    stack: |-
      Error: test exited without ending
         ...
  ...
not ok 3 test exited without ending
  ---
    operator: fail
    stack: |-
      Error: test exited without ending
    ...
  ...
```
I had 3 tests, the first fails and the other automatically fail. If I change the order than, the second passes and the other's fail.

**Comment by ljharb** [comment]
I mean `node path/to/file` - the `tape` runner is only needed if you want to run multiple files at once.

**Comment by Senneseph** [comment]
I know this issue is a little old, but I ran into the same problem and I think I know the root cause.

In your first test, there is no`.catch()` for the promise you get from `fetch`.

So the error message you're getting is not from `tape` but from `node`: `(node:34108) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): #<Object>`

I had to add `try / catch` around my `await`ed function calls to get `tape` running normally again.
